test/api/units_test.rb

```diff
@@ -123,30 +147,67 @@ def assert_tutorial_model_response(response, expected)
     assert_equal response[:abbrev], expected[:abbrev]
   end
 
+  # Test POST for rollover unit should success when missing teaching_period_id
+  def test_rollover_unit_post_success_missing_teaching_period_id
+    unit = Unit.create!(
+      code: "COS10001_TEST",
+      name: "Introduction to Programming",
+      start_date: '2016-05-14T00:00:00.000Z',
+      end_date: '2017-05-14T00:00:00.000Z',
+      description: 'testing'
+    )
+    data_to_post = add_auth_token(
+      start_date: '2016-05-14T00:00:00.000Z',
+      end_date: '2017-05-14T00:00:00.000Z'
+    )
+
+    post_json "/api/units/#{unit.id}/rollover.json", data_to_post
+
+    assert_equal unit.code, last_response_body['code']
+    assert_equal unit.name, last_response_body['name']
+    assert_equal unit.description, last_response_body['description']
+  end
+
+  # Test POST for rollover unit should success when exist teaching_period_id
+  def test_rollover_unit_post_success_exist_teaching_period_id
+    unit = Unit.create!(
+      code: "COS10002_TEST",
+      name: "Introduction to Programming",
+      start_date: '2016-05-14T00:00:00.000Z',
+      end_date: '2017-05-14T00:00:00.000Z',
+      description: 'testing'
+    )
+    data_to_post = add_auth_token(
+      teaching_period_id: 1,
+    )
+
+    post_json "/api/units/#{unit.id}/rollover.json", data_to_post
+
+    assert_equal unit.code, last_response_body['code']
+    assert_equal unit.name, last_response_body['name']
+    assert_equal unit.description, last_response_body['description']
+  end
 
-  def test_addtutorial_to_unit
```

Comment:
Why is this test removed?

Reply:
Because newest code on branch development removed the api